### DISCUSSION TITLE
How to retrieve and set cookies on the server-side?

### DISCUSSION THREAD

**Discussion by geng130127** [vikejs/vike#1020] [Discussion] [answered]
"I want to retrieve a cookie in _default.page.server, how can I do that?"

**Comment by brillout** [comment, marked as answer]
https://vite-plugin-ssr.com/pageContext#custom

**Reply by geng130127** [reply]
Is it set and obtained in the hook onBeforeRender() and render()? But I tried, can't get it

**Reply by brillout** [reply]
https://github.com/brillout/vite-plugin-ssr/commit/1ff94aeb5b85b12ffd6eb362dc339f17aa568302
